core/src/main/groovy/noe/tomcat/configure/ConnectorTomcatAbstract.groovy

```diff
@@ -103,4 +109,19 @@ abstract public class ConnectorTomcatAbstract<E extends ConnectorTomcatAbstract>
     return (E) this
   }
 
+  public E setUpgradeProtocol(UpgradeProtocolTomcat upgradeProtocol = new UpgradeProtocolTomcat()) {
+    this.upgradeProtocol = upgradeProtocol
+    return (E) this
+  }
+
+  public E setUpgradeProtocolClassName(String className) {
```

Comment:
Let's leave up to the client to put an instance of upgradeProtocol initialized properly. I wouldn't recommend exposing UpgradeProtocol attributes configuration into the Connector interface. Consider the situation that you need to add other attributes or even another element (and its attributes) - just KISS (keep it stupid simple) - please have [single responsibility principle ](https://en.wikipedia.org/wiki/Single-responsibility_principle) on mind.